File: activity/20200805/RabbitLee/Find_Smallest_Letter_Greater_Than_Target.py
class Solution:
    def nextGreatestLetter(self, letters: List[str], target: str) -> str:
        size = 26
        letters = set(letters)
        for i in range(1, size):
            next_ord = ord(target) + i
            letter = chr(next_ord if next_ord <= ord('z') else next_ord - size)
            if letter in letters:
                return letter

File: activity/20200805/scau-flyingbird/Find_Smallest_Letter_Greater_Than_Target.go
func nextGreatestLetter(letters []byte, target byte) byte {
	length := len(letters)
	left, right := 0, length-1
	for left <= right {
		mid := left + (right-left)/2
		if letters[mid] <= target {
			left = mid + 1
		} else {
			right = mid - 1
		}
	}
	return letters[left%length]
}
